### DISCUSSION TITLE
Location Filter Threshold / detections filter #Problem

### DISCUSSION THREAD

**Discussion by MueJosh** [tphakala/birdnet-go#94] [General]
## SOLVED: https://github.com/tphakala/birdnet-go/issues/96#issuecomment-2032228280

Hey, 

I dug up some stuff from BirdNET 
It is really important that the user of BirdNET-Go can set the locationFilterThreshold for themselves! 

See line 62
[BirdNET-Analyzer; config.py](https://github.com/kahst/BirdNET-Analyzer/blob/main/config.py)

I think you defined it in Line 33 
[rangefilter.go](https://github.com/tphakala/birdnet-go/blob/main/internal/birdnet/rangefilter.go)

If the threshold is fixed (like it is right now) I can't detect certain species that are definitely present on my location. 

On BirdNET-Pi (Nachtzuster Fork- but its basically the same) it looks like this: 
![image](https://github.com/tphakala/birdnet-go/assets/145150281/73c2cb24-adb5-438a-ab22-3b673becfbcb)
The link from the photo
[here](https://github.com/mcguirepr89/BirdNET-Pi/wiki/BirdNET-Pi:-some-theory-on-classification-&-some-practical-hints)

There is also a small species list tester- so i know where i set my threshold 
For the extreme i set it to 0.95 - see Picture below
![image](https://github.com/tphakala/birdnet-go/assets/145150281/0f3150ce-f838-4364-af73-ff6563964299)
And for my location I have it set to 0.0018 so i can detect most owls in my region. 
As you can see the list gets MUCH longer 
![image](https://github.com/tphakala/birdnet-go/assets/145150281/a78e38bf-96f6-41c4-8fd7-823f7e5f38f0)

## Otherwise a lot of birds don't get detected even tho they are present, calling and are in the birdnet library.

**Comment by tphakala** [comment]
Please create an issue with enhancement tag and I will add this as configurable setting

**Reply by MueJosh** [reply]
[Done](https://github.com/tphakala/birdnet-go/issues/96)
